lib/measured/measurable.rb

```diff
@@ -1,6 +1,8 @@
 class Measured::Measurable < Numeric
   include Measured::Arithmetic
 
+  PARSE_REGEX = /\A\s*([0-9.]+)\s*([a-zA-Z_-]+)\s*\Z/
```

Comment:
`Unit` does [this](https://github.com/Shopify/measured/blob/2a05bfc90ea95a29fc11cd693b20c9e84eec2be7/lib/measured/unit.rb#L66-L71), for conversion values when building a unit system, so we should probably share the parsing code between them.

Two cases missing atm: negative numbers (e.g., -1°C) and unit names with spaces (e.g., nautical mile).